### ISSUE TITLE
Weaver fails if assembly contains an iterator method with ValueTuple return type that has specified member names

### ISSUE THREAD

**Issue by erikvk** [Starcounter/Home#445]
Starcounter version: `2.3.2.930`.

### Issue type

- [x] Bug
- [ ] Feature request
- [ ] Suggestion
- [ ] Question
- [ ] Cannot reproduce
- [ ] Urgent

### Issue description

If a Starcounter assembly contains an iterator method with a return type `IEnumerable<T>` where `T` is a `ValueTuple` with named tuple members (for example `(string S, int I)`), the Weaver will fail while processing that assembly, giving the following error message:

```
ScErrWeaverFailedLoadFile (SCERR2140): The weaver failed to load a binary user code file. File: C:\Users\erik\DRTB\Mopedo DSP\bin\Debug\Mopedo.dll
Version: 2.3.2.930.
Help page: https://docs.starcounter.io/v/2.3.1/?q=SCERR2140.
```  

### Reproduction code snippet

```cs
public class Program
{
    public static void Main() { }
    
    private static IEnumerable<(int Int, TimeSpan TimeSpan)> GetThings()
    {
        yield return (default, default);
    }
}
```

Note, however, that this works:

```cs
public class Program
{
    public static void Main() { }
    
    private static IEnumerable<(int, TimeSpan)> GetThings()
    {
        yield return (default, default);
    }
}
```

**Comment by erikvk** [comment]
**cc** @miyconst @per-samuelsson

**Comment by miyconst** [comment]
@erikvk thank you for the detailed report and the investigation!

**Comment by miyconst** [comment]
Closing as won't fix in Starcounter 2.4 and not applicable in Starcounter 3.0.
